### PR TITLE
Added header validation library

### PR DESCRIPTION
**Pull request by msk-**
See: https://github.com/mojaloop/account-lookup-service/pull/41#discussion_r318604095 and the enclosing PR for motivation and some context (I expect readers of this PR are familiar, hence my lazy non-reproduction here).

Notes:
- Haven't written a test for the plugin (but have tested it!). I intend to, but it's near the end of my day now- so don't merge this until I have.
- Similarly, I'll address other test coverage tomorrow
- Basically, I expect some feedback on a few things, so I'm seeking it now

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
   "devDependencies": {
     "@hapi/hapi": "18.3.2",
     "@hapi/joi": "15.1.1",
+    "chance": "^1.0.18",
```
Comment (rmothilal) [inline package.json]: please remove caret

Comment (rmothilal) [reply package.json]: and updated version of the package.json

File: src/util/headerValidation/.gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# The tests store failures in a file for easier inspection- this is because test failures for
+# fuzzing can be large
+testfail*/
```
Comment (rmothilal) [inline src/util/headerValidation/.gitignore]: why is this .gitignore here?

Comment (msk-) [reply src/util/headerValidation/.gitignore]: Because the header validation code is fuzz tested the output can be quite large, so failures are printed to a file also, to make it a bit easier to inspect. This gitignore file refers to those output files.

File: package.json (hunk 1 above)
Comment (msk-) [reply package.json]:
```suggestion
```

Comment (msk-) [reply package.json]: Welp I removed it then had to re-add. What do you mean "and updated version of the package.json"?
